### ISSUE TITLE
Produce Microcontroller Comms Protocol Wiki Page

### ISSUE THREAD

**Issue by arandell93** [LakeMaps/microcontrollers#1]
I've started development of a wiki page for the comms protocol.

I need @whymarrh and @FifoIronton to review what I've done so far and provide feedback.

We'll use this issue to bring the comms protocol documentation to a point where we can produce an API from it on both the Arduino and Pi/PC

**Comment by whymarrh** [comment]
A few quick thoughts:
1. I think the page can simply be 'API Documentation'
2. The first title should be a 2nd heading (`##`), where the page title is the first heading
3. Needs moar hyperlinks, you can mouseover a title to get an anchor:

![2016-11-17 22 54 30](https://cloud.githubusercontent.com/assets/1623628/20416324/da7f5980-ad18-11e6-8266-a94852506652.gif)

**Comment by whymarrh** [comment]
It might make sense to have separate pages for each micro-controller and a top-level page with links

**Comment by arandell93** [comment]
> Needs moar hyperlinks, you can mouseover a title to get an anchor:

What do I do with said anchor? Do I make a pretty TOC at the top manually or make one of those custom sidebar things with links to each section?

![image](https://cloud.githubusercontent.com/assets/10678515/20434083/e56ac330-ad80-11e6-8977-e362a8d9a57b.png)

**Comment by whymarrh** [comment]
ヽ(。_°)ノ I'd go for making a TOC at the top manually. If you're looking for inspiration, you can borrow ideas from [Projects with great wikis](https://github.com/showcases/projects-with-great-wikis) (e.g. the [Hystrix Configuration page](https://github.com/Netflix/Hystrix/wiki/Configuration)).

**Comment by arandell93** [comment]
Am I [doin it rite](https://github.com/LakeMaps/Arduino/wiki/API-Documentation)?

**Comment by arandell93** [comment]
@whymarrh I have finished the [Wireless Module Command Page](https://github.com/LakeMaps/Arduino/wiki/Wireless-Communication-Microcontroller-API) but I came across a design inconsistency that I would like your input on.

On the [Propulsion Module Command Page](https://github.com/LakeMaps/Arduino/wiki/Propulsion-Microcontroller-API#command-0x11--set-motor-controller-configuration) I describe the set/get configuration as being all inclusive; that is a single command is used to set or read all configuration parameters at once. For the motor controller, this is not a big deal because there are not too many. In order to provide a selectivity function, I added a word (16 bits) at the beginning which would mask out the parameters you wish to change.

However, with the [Wireless Module](https://github.com/LakeMaps/Arduino/wiki/Wireless-Communication-Microcontroller-API#command-0x01--set-wireless-module-configuration) there are a lot more registers so it didn't make sense to do them all at once, so I opted to use a command that only accesses one at a time.

While both are equally valid methods of setting/reading the configuration, I would like to be consistent and as such change the Propulsion Module commands to also only address one at a time. 
Thoughts?
